services/app-api/resolvers/updateHealthPlanFormData.test.ts

```diff
@@ -0,0 +1,338 @@
+            context: {
+                user: {
+                    name: 'Aang',
+                    state_code: 'VA',
+                    role: 'STATE_USER',
+                    email: 'aang@va.gov',
+                },
+            },
+        })
+
+        const updateResult = await otherUserServer.executeOperation({
+            query: UPDATE_HEALTH_PLAN_FORM_DATA,
+            variables: {
+                input: {
+                    submissionID: createdDraft.id,
+                    healthPlanFormData: updatedB64,
+                },
+            },
+        })
+
+        expect(updateResult.errors).toBeDefined()
+        if (updateResult.errors === undefined) {
+            throw new Error('type narrow')
+        }
+
+        expect(updateResult.errors[0].extensions?.code).toBe('FORBIDDEN')
+        expect(updateResult.errors[0].message).toBe(
+            'user not authorized to fetch data from a different state'
+        )
+    })
+
+    it('errors if the payload isnt valid', async () => {
+        const server = await constructTestPostgresServer()
+
+        const createdDraft = await createTestSubmission2(server)
+
+        const formData = 'not-valid-proto'
+
+        const updateResult = await server.executeOperation({
+            query: UPDATE_HEALTH_PLAN_FORM_DATA,
+            variables: {
+                input: {
+                    submissionID: createdDraft.id,
+                    healthPlanFormData: formData,
+                },
+            },
+        })
+
+        expect(updateResult.errors).toBeDefined()
+        if (updateResult.errors === undefined) {
+            throw new Error('type narrow')
+        }
+
+        expect(updateResult.errors[0].extensions?.code).toBe('BAD_USER_INPUT')
+        expect(updateResult.errors[0].message).toContain(
+            'Failed to parse out form data in request'
+        )
+    })
+
+    it('errors if the payload is submitted', async () => {
+        const server = await constructTestPostgresServer()
+
+        const createdDraft = await createTestSubmission2(server)
+
+        const stateSubmission = basicStateSubmission()
+
+        const formData = domainToBase64(stateSubmission)
+
+        const updateResult = await server.executeOperation({
+            query: UPDATE_HEALTH_PLAN_FORM_DATA,
+            variables: {
+                input: {
+                    submissionID: createdDraft.id,
+                    healthPlanFormData: formData,
+                },
+            },
+        })
+
+        expect(updateResult.errors).toBeDefined()
+        if (updateResult.errors === undefined) {
+            throw new Error('type narrow')
+        }
+
+        expect(updateResult.errors[0].extensions?.code).toBe('BAD_USER_INPUT')
+        expect(updateResult.errors[0].message).toContain(
+            'Attempted to update with a StateSubmission'
+        )
+    })
+
+    it('errors if the payload is already submitted', async () => {
```

Comment:
I _think_ this is a duplicate test. Hmm trying to think.. was it supposed to maybe a test an unlocked and resubmitted submission instead? Or maybe just accidental

Reply:
you are right! but I think I’m missing a test for if the whole submission2 is in a submitted state